content/systems/filecoin_blockchain/struct/block/_index.md

```diff
@@ -2,25 +2,109 @@
 
-{{< embed src="election.id" lang="go" >}}
+```go
+type FullBlock struct {
+	Header        *BlockHeader
+	BlsMessages   []*Message
+	SecpkMessages []*SignedMessage
+}
+```
+
+> **Note:** A block is functionally the same as a block header in the Filecoin protocol. While a block header contains Merkle links to the full system state, messages, and message receipts, a block can be thought of as the full set of this information (not just the Merkle roots, but rather the full data of the state tree, message tree, receipts tree, etc.). Because a full block is large in size, the Filecoin blockchain consists of block headers rather than full blocks. We often use the terms `block` and `block header` interchangeably.
+
+A `BlockHeader` is a canonical representation of a block. BlockHeaders are propagated between miner nodes. From the blockcheader message, a miner has all the required information to apply the associated `FullBlock`'s state and update the chain. In order to be able to do this, the minimum set of information items that need to be included in the `BlockHeader` are shown below and include among others: the miner's address, the Ticket, the [Proof of SpaceTime](post), the CID of the parents where this block evolved from in the IPLD DAG, as well as the messages' own CID.
+
+The Lotus implementation of the block header can be found [here](https://github.com/filecoin-project/lotus/blob/master/chain/types/blockheader.go). It has the following `struct`s:
+
+```go
+type BlockHeader struct {
+	Miner address.Address // 0
+	Ticket *Ticket // 1
+	ElectionProof *ElectionProof // 2
+	BeaconEntries []BeaconEntry // 3
+	WinPoStProof []proof.PoStProof // 4
+	Parents []cid.Cid // 5
+	ParentWeight BigInt // 6
+	Height abi.ChainEpoch // 7
+	ParentStateRoot cid.Cid // 8
+	ParentMessageReceipts cid.Cid // 9
+	Messages cid.Cid // 10
+	BLSAggregate *crypto.Signature // 11
+	Timestamp uint64 // 12
+	BlockSig *crypto.Signature // 13
+	ForkSignaling uint64 // 14
+	// ParentBaseFee is the base fee after executing parent tipset
+	ParentBaseFee abi.TokenAmount // 15
+	// internal
+	validated bool // true if the signature has been validated
+}
+
+type Ticket struct {
+	VRFProof []byte
+}
+
+type ElectionProof struct {
+	VRFProof []byte
+}
+
+type BeaconEntry struct {
+	Round uint64
+	Data  []byte
+}
+```
+
+The `BlockHeader` structure has to refer to the TicketWinner of the current round ++ which eensures the correct winner is passed to [ChainSync](chainsync).
+
+```go
+func IsTicketWinner(vrfTicket []byte, mypow BigInt, totpow BigInt) bool
+```
+
+The `Message` structure has to include the source and destination addresses, a `Nonce` and the `GasPrice`.
+
+The Lotus implementation of the message can be found [here](https://github.com/filecoin-project/lotus/blob/master/chain/types/message.go). It has the following structure:
```

Comment:
Ideally at some point in the future it would be great if (by some magic completely unknown to me) we could pull the Go definition from that link and display it here dynamically instead of having to manually maintain this.